config/testdata/kubernetes_kube_config_without_apiserver.good.yml

```diff
@@ -0,0 +1,5 @@
+scrape_configs:
+- job_name: prometheus
+  kubernetes_sd_configs:
+  - role: endpoints
+    kube_config: ~/.kube/config
```

Comment:
Does Prometheus expand that? I am surprised.

Reply:
yeah, if kube_config specified ,kube config contains all information that need to connect api-server
I am test it ,and it work fine.

Reply:
my test config
```yaml
scrape_configs:
  # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
  - job_name: 'prometheus'

    # metrics_path defaults to '/metrics'
    # scheme defaults to 'http'.

    kubernetes_sd_configs:
    - role: pod
      kube_config: /Users/edz/.kube/config
```

Reply:
So potentially `~/` will not work. Could you test with `~/.kube/config` ?

Reply:
I am sorry, i test it , it doesn't  work, I add comment to the doc, it should be absolute path ,or relative path to prometheus'config